internal/controller/log.go

```diff
@@ -48,15 +51,25 @@ func (d *DeployLog) LogDeploy(timeStamp time.Time, podName string, podIP string,
 		Reason:           reason,
 	}
 
-	d.Entries = append(d.Entries, newEntry)
+	if len(d.entries) >= d.maxEntries {
+		// Pull elements off the front of the slice to make sure that the newly appended record is under the max record value.
+		d.entries = d.entries[(len(d.entries)-d.maxEntries)+1:]
```

Comment:
Hmmm, the only possible case here is `len(d.entries) == d.maxEntries` right ? 
So `len(d.entries) - d.maxEntries == 0`  at any point ? 

Then this is correct ?
```suggestion
		d.entries = d.entries[1:]
```

Reply:
I wanted to cover a theoretical edge case where somehow 2 entries got added before the slice was pruned, so it could have +2 entries etc.